### PR TITLE
Creation of release candicate v0.10.0-rc

### LINKED ISSUE #235: Release v0.10.0 - Release Candidate due Nov 24th
**Problem description**

Multiple operators are looking for a release of v0.10.0 to be able to implement it within Q1/2024 (potentially for announcements at MWCB). The main difference of v0.10.0 towards will be the new event structure with CloudEvents as decided and document within the CAMARA Design Guidelines. 

The release candidate for v0.10.0 should be ** created at end of next week (Friday Nov 23rd) ** to have enough time to validate the release candidate by implementing operators and do the actual release of v0.10.0 at latest before year end break.

**Expected action**

The following PRs MUST be included in the release candidate v0.10.0

- [ ] [#216 Extend session duration](https://github.com/camaraproject/QualityOnDemand/pull/216)
- [ ] [#224 Align event notification with CloudEvents spec](https://github.com/camaraproject/QualityOnDemand/pull/224)
- [ ] [#233 Alignment with changes in https://github.com/camaraproject/Template_Lead_Repository on test definitions](https://github.com/camaraproject/QualityOnDemand/pull/233)
- [ ] Yet to be prepared PR with the update of CHANGELOG.md, README.md and the change of the version number within the OAS file from v0.10.0-wip to v0.10.0-rc. The merge commit of this PR will be the tagged as the release candidate.

The following PRs COULD be included within the release candidate if finalized in time

- [ ] [#228 Add GET /sessions endpoint for retrieving all active sessions of authenticated user](https://github.com/camaraproject/QualityOnDemand/pull/228)
- [ ] Yet to be done PR for issue [#230 Single IPv6 addresses within device](https://github.com/camaraproject/QualityOnDemand/issues/230)

**Additional context**

The release candidate will be the starting point for implementations. Release to be expected about 2 or 4 weeks later.
Between release candidate and release only bug fixes and updates of documentation will be done on the API spec. In case of changes to the OAS file there might be second release candidate ("rc2").

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

Add one of the following kinds:

* documentation
* subproject management

#### What this PR does / why we need it:

The merge of this PR will be the target commit for the first release candidate of v0.10.0, getting the tag `v0.10.0-rc`

#### Which issue(s) this PR fixes:

Fixes #235 

#### Special notes for reviewers:

Currently draft PR, as the final information about the PRs included in the release candidate are not yet available.

#### Changelog input

NA

### CODE REVIEW

**Comment by hdamker** [comment]
Updated CHANGELOG.md to reflect the merge of #216, #224 and #233.

Still waiting for #237 as this will correct an implementation issue with the IPv6 address pattern (see discussion in #230).

**Review by sfnuser — APPROVED** [review]
Looks good.

**Review by eric-murray — APPROVED** [review]
All looks good to me. I'd even expect all the links to work when the `release-0.10.0-rc` branch is created.

I had thought PR #228 was still in scope. I did request some changes to that, but if agreed, then I'm happy to include that in the release as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +13,57 @@ Version numbers 0.2.x to 0.7.x were intentionally not used to avoid conflicts wi
...
+
+- API definition **with inline documentation**:
+  - OpenAPI [YAML spec file](https://github.com/camaraproject/QualityOnDemand/blob/release-0.10.0-rc/code/API_definitions/qod-api.yaml)
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/release-0.10.0-rc/code/API_definitions/qod-api.yaml&nocors)
+  - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/release-0.10.0-rc/code/API_definitions/qod-api.yaml)
+
+## Please note:
+
+- **This release will contain significant changes compared to v0.9.0, and it is not backward compatible**
+  - Especially ...
```
Comment (jlurien) [inline CHANGELOG.md:28]: This seems incomplete

Comment (RandyLevensalor) [reply CHANGELOG.md:28]: Once we have the open PRs merged, we should list the changes that break backwards compatibility.  We've had changes in those today.

**Review by jlurien — APPROVED** [review]

**Review by RandyLevensalor — APPROVED** [review]

**Pull request merged**